site/e2e/helpers.ts

```diff
@@ -490,6 +490,16 @@ const createTemplateVersionTar = async (
             token: randomUUID(),
             ...agent,
           } as Agent;
+
+          try {
+            Agent.encode(agentResource);
+          } catch (e) {
+            throw new Error(
+              "agentResource encode failed, missing defaults? " + e,
```

Comment:
One gotcha with JS errors: their default stringification method only grabs the error's `name` and `message` properties – the entire stack gets lost (partially because `stack` technically is a non-standard property, even though it's basically added everywhere)

I would go for something like this:
```ts
throw new Error(
  `agentResource encode failed, missing defaults?\n${e}\n${e.stack ?? ""}`
);
```

Reply:
I was looking to avoid the double stack trace (hence discarding the original one since we still create one at `new Error`), but I agree that the original stack can also be useful, I'll make the change 👍🏻